src/main/kotlin/org/rust/ide/annotator/RsHighlightingAnnotator.kt

```diff
@@ -92,6 +110,8 @@ class RsHighlightingAnnotator : AnnotatorBase() {
             } else {
                 null
             }
+            parent is RsMacroBinding -> RsColor.MACRO
+            parent is RsMacroReference -> RsColor.MACRO
```

Comment:
Looks like this change affects definitions of macro variables

<img width="552" alt="Screen Shot 2022-06-08 at 18 41 04" src="https://user-images.githubusercontent.com/2539310/172659825-5957fac6-0839-4eed-b0d6-6a1c6e9645f7.png">

Reply:
Let's look at these two branches separately:
- `parent is RsMacroReference -> RsColor.MACRO` - this code is never called at all since `RsMacroReference` case is handled by `parent is RsReferenceElement && parent !is RsModDeclItem && ...`  branch. That's why currently meta variable references are highlighted differently. So we can just drop it
- `parent is RsMacroBinding -> RsColor.MACRO` is redundant here as well, since we have the same code for `RsMacroBinding` in `colorFor` function

It's a technical view why this code here should be dropped/moved.

About "macro-machinery-specific token" and `FUNCTION` highlighting - all meta variables already have specific highlighting, so you can easily find them in macros, and I consider that you want to have different highlighting for them regardless of highlighting for other tokens. So let's keep this highlighting as is for now as we discussed in https://github.com/intellij-rust/intellij-rust/pull/8868#discussion_r892429205

Reply:
I've tried removing those now, and it makes it look like this:

![image](https://user-images.githubusercontent.com/7755349/172704484-218193f0-0322-4049-b53e-6a7460a004b7.png)

In my opinion this goes counter to the goal of this PR. Here, the `meta` after the colon looks as if it's a regular Rust token and not a macro thing. Likewise `item`.

Reply:
Ok, I've implemented new RsColors--then it does look better. Will polish and push later.

Reply:
> I've implemented new RsColors now.

Let's keep the first plan to restrict the scope of this PR and not to add new features to it. Otherwise, it will lead to new changes, new test failures, etc.

> Here, the `meta` after the colon looks as if it's a regular Rust token (which will written out as-is by the macro expansion) and not like a macro machinery thing. Likewise `item`.

Yeah, I see. It's because of [this small hack](https://github.com/intellij-rust/intellij-rust/blob/fd79f463cd5ae9f6b5eebaad16ce193db1b68d18/src/main/kotlin/org/rust/ide/annotator/RsHighlightingAnnotator.kt#L61-L64) not to differ identifiers as meta item names and their types. So I think we just can drop this "hack" and move the necessary logic into `highlightIdentifier` method. It allows us to distinguish both cases

Reply:
I've reverted the extended color fix for now.

The current version looks okay to me at runtime, except for nested macro definitions `cfg_if! { macro_rules! ... }`--which are maybe out of scope anyway.